packages/moonstone/VideoPlayer/VideoPlayer.js

```diff
@@ -755,9 +764,17 @@ const VideoPlayerBase = class extends React.Component {
 			this.calculateMaxComponentCount();
 		}
 
-		const {source} = this.props;
+		const {source, preloadSource} = this.props;
 		const {source: nextSource} = nextProps;
 
+		if (preloadSource && compareSources(preloadSource, nextSource)) {
+			this.preloadSourcePlaying = !this.preloadSourcePlaying;
+			const currentVideoSource = this.video;
+			this.video = this.preloadVideo;
+			this.preloadVideo = currentVideoSource;
```

Comment:
Why swap the current video to preload?  And, do we need to call `load()` on it if it was already loaded?

Reply:
What happens when the preload video is different from the previous preload video?  Do we need to handle that case?

Reply:
The reason I swap the current video to preload is because of the refs. Refs don't re run on render so we have to swap it out then call load on it. This way we can reuse the ref for loading the next video.

Reply:
Also I reworked the loaded stuff. I think we should only call it after componentDidUpdate. This way it will load after the rendering. I think that makes most sense.